azurerm/internal/services/apimanagement/api_management_diagnostic_resource.go

```diff
@@ -95,6 +144,70 @@ func resourceApiManagementDiagnosticCreateUpdate(d *schema.ResourceData, meta in
 		},
 	}
 
+	if samplingPercentage, ok := d.GetOk("sampling_percentage"); ok {
+		parameters.Sampling = &apimanagement.SamplingSettings{
+			SamplingType: apimanagement.Fixed,
+			Percentage:   utils.Float(samplingPercentage.(float64)),
+		}
+	} else {
+		parameters.Sampling = nil
+	}
+
+	if alwaysLogErrors, ok := d.GetOk("always_log_errors"); ok && alwaysLogErrors.(bool) {
+		parameters.AlwaysLog = apimanagement.AllErrors
+	}
+
+	if verbosity, ok := d.GetOk("verbosity"); ok {
+		switch verbosity.(string) {
+		case string(apimanagement.Verbose):
+			parameters.Verbosity = apimanagement.Verbose
+		case string(apimanagement.Information):
+			parameters.Verbosity = apimanagement.Information
+		case string(apimanagement.Error):
+			parameters.Verbosity = apimanagement.Error
+		}
```

Comment:
I think we can change this to
```suggestion
		parameters.Verbosity = apimanagement.Verbosity(verbosity.(string)) 
```

Reply:
I've made the update here as well as in `api_management_api_diagnostic_resource.go` where this code was originally copied from.